cogs/roles.py

```diff
@@ -257,8 +264,238 @@ async def create_role(self, ctx, *, role: Optional[str] = None):
         await ctx.guild.create_role(name=role, reason="Created with Canary")
         await ctx.send("Role created successfully.")
 
-    @commands.command()
-    async def inchannel(self, ctx):
+    def _save_existing_roles(self,
+                             user: discord.Member,
+                             penalty: bool = False):
+        table = "PenaltyUsers" if penalty else "PreviousRoles"
+
+        roles_id = [
+            role.id for role in user.roles
+            if role.name not in ("@everyone", self.penalty_role)
+        ]
+
+        if not roles_id:
+            return
+
+        conn = sqlite3.connect(self.bot.config.db_path)
+        try:
+            c = conn.cursor()
+            # store roles as a string of IDs separated by spaces
+            t = (user.id, " ".join(str(e) for e in roles_id))
+            c.execute(f"REPLACE INTO {table} VALUES (?, ?)", t)
```

Comment:
If a user gets muted but has no roles, they wont get added to the penalty table

Reply:
9dd2412